### PR TITLE
Added additional logger statements for issue#24

### PR DESCRIPTION
**Pull request by aditya566**

### CODE REVIEW

File: src/pycatcher/diagnostics.py [hunk 1]
```diff
@@ -183,11 +183,11 @@ def conduct_stationarity_check(df):
     # Perform Augmented Dickey-Fuller test
     adf_result = adfuller(df_pandas.iloc[:, -1])
 
-    print('ADF Statistic: %f' % adf_result[0])
-    print('p-value: %f' % adf_result[1])
-    print('Critical Values:')
+    logger.info("ADF Statistic: ", round(adf_result[0], 6))
```
Comment (jpamarthi) [inline src/pycatcher/diagnostics.py:186]: Lets remove this decimal since the original code got updated.

**Comment by jpamarthi** [comment]
@aseemanand Could you review this when you get a chance?

**Review by aseemanand — CHANGES_REQUESTED** [review]
logger.info for ADF test will need formatters (%d, %f etc). We see error while running conduct_stationarity_check.

2024-11-24 10:04:58,416 - INFO - Starting ADF stationarity check
--- Logging error ---
Traceback (most recent call last):
  File "[/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 1160](http://localhost:8888/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py#line=1159), in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "[/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 999](http://localhost:8888/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py#line=998), in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "[/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 703](http://localhost:8888/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py#line=702), in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "[/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 392](http://localhost:8888/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py#line=391), in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel_launcher.py", line 18](http://localhost:8888/venv/lib/python3.12/site-packages/ipykernel_launcher.py#line=17), in <module>
    app.launch_new_instance()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelapp.py", line 739, in start
    self.io_loop.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/tornado/platform/asyncio.py", line 205, in start
    self.asyncio_loop.run_forever()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 1987, in _run_once
    handle._run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 545, in dispatch_queue
    await self.process_one()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 534, in process_one
    await dispatch(*args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 437, in dispatch_shell
    await result
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 362, in execute_request
    await super().execute_request(stream, ident, parent)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 778, in execute_request
    reply_content = await reply_content
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 449](http://localhost:8888/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py#line=448), in do_execute
    res = shell.run_cell(
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/zmqshell.py", line 549, in run_cell
    return super().run_cell(*args, **kwargs)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3048, in run_cell
    result = self._run_cell(
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3103, in _run_cell
    result = runner(coro)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/async_helpers.py", line 129, in _pseudo_sync_runner
    coro.send(None)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3308, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3490](http://localhost:8888/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py#line=3489), in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3550, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "/var/folders/74/_c6jvy7n6vn0tkrvg80t_1gh0000gp/T/ipykernel_92212/755627028.py", line 1, in <module>
    conduct_stationarity_check(df_tsa)
  File "/Users/aseem/Documents/GitHub/outlier_detection/src/pycatcher/diagnostics.py", line 186, in conduct_stationarity_check
    logger.info("ADF Statistic: ", adf_result[0])
Message: 'ADF Statistic: '
Arguments: (np.float64(-1.8218908632694801),)
--- Logging error ---
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 1160, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 703, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 392, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel_launcher.py", line 18, in <module>
    app.launch_new_instance()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelapp.py", line 739, in start
    self.io_loop.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/tornado/platform/asyncio.py", line 205, in start
    self.asyncio_loop.run_forever()
  File "[/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 641](http://localhost:8888/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py#line=640), in run_forever
    self._run_once()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 1987, in _run_once
    handle._run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 545, in dispatch_queue
    await self.process_one()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 534, in process_one
    await dispatch(*args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 437, in dispatch_shell
    await result
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 362, in execute_request
    await super().execute_request(stream, ident, parent)
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 778](http://localhost:8888/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py#line=777), in execute_request
    reply_content = await reply_content
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 449, in do_execute
    res = shell.run_cell(
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/zmqshell.py", line 549](http://localhost:8888/venv/lib/python3.12/site-packages/ipykernel/zmqshell.py#line=548), in run_cell
    return super().run_cell(*args, **kwargs)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3048, in run_cell
    result = self._run_cell(
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3103, in _run_cell
    result = runner(coro)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/async_helpers.py", line 129, in _pseudo_sync_runner
    coro.send(None)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3308, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3490, in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3550, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "[/var/folders/74/_c6jvy7n6vn0tkrvg80t_1gh0000gp/T/ipykernel_92212/755627028.py", line 1](http://localhost:8888/var/folders/74/_c6jvy7n6vn0tkrvg80t_1gh0000gp/T/ipykernel_92212/755627028.py#line=0), in <module>
    conduct_stationarity_check(df_tsa)
  File "/Users/aseem/Documents/GitHub/outlier_detection/src/pycatcher/diagnostics.py", line 187, in conduct_stationarity_check
    logger.info("p-value: ", adf_result[1])
Message: 'p-value: '
Arguments: (np.float64(0.36964891592783444),)
2024-11-24 10:04:58,491 - INFO - Critical Values:
--- Logging error ---
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 1160, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 703, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 392, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel_launcher.py", line 18, in <module>
    app.launch_new_instance()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelapp.py", line 739, in start
    self.io_loop.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/tornado/platform/asyncio.py", line 205, in start
    self.asyncio_loop.run_forever()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 1987, in _run_once
    handle._run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 545, in dispatch_queue
    await self.process_one()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 534, in process_one
    await dispatch(*args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 437, in dispatch_shell
    await result
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 362, in execute_request
    await super().execute_request(stream, ident, parent)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 778, in execute_request
    reply_content = await reply_content
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 449, in do_execute
    res = shell.run_cell(
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/zmqshell.py", line 549](http://localhost:8888/venv/lib/python3.12/site-packages/ipykernel/zmqshell.py#line=548), in run_cell
    return super().run_cell(*args, **kwargs)
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3048](http://localhost:8888/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py#line=3047), in run_cell
    result = self._run_cell(
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3103, in _run_cell
    result = runner(coro)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/async_helpers.py", line 129, in _pseudo_sync_runner
    coro.send(None)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3308, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3490, in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3550](http://localhost:8888/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py#line=3549), in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "/var/folders/74/_c6jvy7n6vn0tkrvg80t_1gh0000gp/T/ipykernel_92212/755627028.py", line 1, in <module>
    conduct_stationarity_check(df_tsa)
  File "/Users/aseem/Documents/GitHub/outlier_detection/src/pycatcher/diagnostics.py", line 190, in conduct_stationarity_check
    logger.info("\t", key, ":", round(value, 3))
Message: '\t'
Arguments: ('1%', ':', np.float64(-3.434))
--- Logging error ---
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 1160, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 703, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 392, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel_launcher.py", line 18, in <module>
    app.launch_new_instance()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelapp.py", line 739, in start
    self.io_loop.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/tornado/platform/asyncio.py", line 205, in start
    self.asyncio_loop.run_forever()
  File "[/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 641](http://localhost:8888/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py#line=640), in run_forever
    self._run_once()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 1987, in _run_once
    handle._run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 545, in dispatch_queue
    await self.process_one()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 534, in process_one
    await dispatch(*args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 437, in dispatch_shell
    await result
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 362](http://localhost:8888/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py#line=361), in execute_request
    await super().execute_request(stream, ident, parent)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 778, in execute_request
    reply_content = await reply_content
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 449, in do_execute
    res = shell.run_cell(
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/zmqshell.py", line 549, in run_cell
    return super().run_cell(*args, **kwargs)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3048, in run_cell
    result = self._run_cell(
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3103, in _run_cell
    result = runner(coro)
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/async_helpers.py", line 129](http://localhost:8888/venv/lib/python3.12/site-packages/IPython/core/async_helpers.py#line=128), in _pseudo_sync_runner
    coro.send(None)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3308, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3490, in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3550, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "/var/folders/74/_c6jvy7n6vn0tkrvg80t_1gh0000gp/T/ipykernel_92212/755627028.py", line 1, in <module>
    conduct_stationarity_check(df_tsa)
  File "/Users/aseem/Documents/GitHub/outlier_detection/src/pycatcher/diagnostics.py", line 190, in conduct_stationarity_check
    logger.info("\t", key, ":", round(value, 3))
Message: '\t'
Arguments: ('5%', ':', np.float64(-2.863))
--- Logging error ---
Traceback (most recent call last):
  File "[/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 1160](http://localhost:8888/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py#line=1159), in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 703, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/logging/__init__.py", line 392, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel_launcher.py", line 18, in <module>
    app.launch_new_instance()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelapp.py", line 739, in start
    self.io_loop.start()
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/tornado/platform/asyncio.py", line 205, in start
    self.asyncio_loop.run_forever()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 1987, in _run_once
    handle._run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 545, in dispatch_queue
    await self.process_one()
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 534](http://localhost:8888/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py#line=533), in process_one
    await dispatch(*args)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 437, in dispatch_shell
    await result
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 362, in execute_request
    await super().execute_request(stream, ident, parent)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/kernelbase.py", line 778, in execute_request
    reply_content = await reply_content
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 449, in do_execute
    res = shell.run_cell(
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/ipykernel/zmqshell.py", line 549, in run_cell
    return super().run_cell(*args, **kwargs)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3048, in run_cell
    result = self._run_cell(
  File "[/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3103](http://localhost:8888/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py#line=3102), in _run_cell
    result = runner(coro)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/async_helpers.py", line 129, in _pseudo_sync_runner
    coro.send(None)
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3308, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3490, in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "/Users/aseem/Documents/GitHub/outlier_detection/venv/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3550, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "/var/folders/74/_c6jvy7n6vn0tkrvg80t_1gh0000gp/T/ipykernel_92212/755627028.py", line 1, in <module>
    conduct_stationarity_check(df_tsa)
  File "/Users/aseem/Documents/GitHub/outlier_detection/src/pycatcher/diagnostics.py", line 190, in conduct_stationarity_check
    logger.info("\t", key, ":", round(value, 3))
Message: '\t'
Arguments: ('10%', ':', np.float64(-2.568))
2024-11-24 10:04:58,492 - INFO - Completed ADF stationarity check

**Review by aseemanand — APPROVED** [review]
LGTM. Thanks.

**Comment by jpamarthi** [comment]
Ah! I thought of testing it before asking for a review again. I am glad that it is working as expected now :)

**Pull request merged**